.github/workflows/code_checks.yml

```diff
@@ -220,17 +285,100 @@ jobs:
         with:
           check_name: Test Results
           comment_mode: off
-          files: Test Results/*.xml
+          files: Test Results Group*/*.xml
           github_token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Check test-failure label
+        if: ${{ needs.tests.result == 'success' && github.event_name == 'pull_request' }}
+        id: check-test-failure-label
+        run: |
+          if gh pr view ${{ github.event.pull_request.number }} --json labels --jq '.labels[].name' | grep -q "^test-failure$"; then
+            echo "has_label=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_label=false" >> $GITHUB_OUTPUT
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Remove Test Failure label
+        if: ${{ needs.tests.result == 'success' && github.event_name == 'pull_request' && steps.check-test-failure-label.outputs.has_label == 'true' }}
+        uses: actions-ecosystem/action-remove-labels@v1
+        with:
+          number: ${{ github.event.pull_request.number }}
+          labels: test-failure
+
+      - name: Download coverage artifacts
+        if: needs.tests.result == 'success'
+        uses: actions/download-artifact@v4
+        with:
+          pattern: simplecov-resultset-*
+          path: .
+
+      - name: Checkout code for Docker setup
+        if: needs.tests.result == 'success'
+        uses: actions/checkout@v5
+        with:
+          path: vets-api
+
+      - name: Copy coverage artifacts to workspace
+        if: needs.tests.result == 'success'
+        run: |
+          echo "=== Copying coverage artifacts into vets-api directory ==="
+          cp -r simplecov-resultset-* vets-api/ || echo "No coverage artifacts to copy"
+          echo "=== Verifying artifacts in vets-api directory ==="
+          ls -la vets-api/simplecov-resultset-* || echo "No simplecov-resultset-* directories in vets-api"
+
+      - name: Login to DockerHub
+        if: needs.tests.result == 'success'
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Set up Docker Buildx
+        if: needs.tests.result == 'success'
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker Image for Coverage
```

Comment:
Unfortunately, I'm not sure how else to aggregate the results (cleanly) without booting Docker again. Since the completion time of the suite is the same as before, I'm marking this optimization out of scope of this PR.